main.go

```diff
@@ -165,54 +174,75 @@ func mainE(config Configs) error {
+	artifacts, err := getArtifacts(gradleProject, started, appPatterns, false)
 	if err != nil {
-		return fmt.Errorf("failed to find apks, error: %v", err)
+		return fmt.Errorf("failed to find artifacts, error: %v", err)
 	}
 
-	if len(apks) == 0 {
-		log.Warnf("No apks found with pattern: %s", config.APKPathPattern)
-		log.Warnf("If you have changed default APK export path in your gradle files then you might need to change APKPathPattern accordingly.")
+	if len(artifacts) == 0 {
+		log.Warnf("No artifacts found with patterns: %s", strings.Join(appPatterns, ", "))
+		log.Warnf("If you have changed default APK, AAB export path in your gradle files then you might need to change AppPathPattern accordingly.")
 		return nil
 	}
 
-	exportedArtifactPaths, err := exportArtifacts(apks, config.DeployDir)
+	exportedArtifactPaths, err := exportArtifacts(artifacts, config.DeployDir)
 	if err != nil {
 		return fmt.Errorf("Failed to export artifact: %v", err)
 	}
 
+	log.Debugf("Found artifacts:\n" + strings.Join(exportedArtifactPaths, "\n"))
+
+	// Remove the wrong build types
+	i := 0
+	for _, path := range exportedArtifactPaths {
+		if filepath.Ext(path) == "."+config.BuildType {
+			exportedArtifactPaths[i] = path
+			i++
+		}
+	}
+	exportedArtifactPaths = exportedArtifactPaths[:i]
+
+	log.Debugf("\nFiltered artifacts:\n" + strings.Join(exportedArtifactPaths, "\n"))
+
 	if len(exportedArtifactPaths) == 0 {
-		return fmt.Errorf("Could not export any APKs")
+		return fmt.Errorf("Could not export any artifacts")
 	}
 
 	lastExportedArtifact := exportedArtifactPaths[len(exportedArtifactPaths)-1]
 
 	fmt.Println()
-	if err := tools.ExportEnvironmentWithEnvman(apkEnvKey, lastExportedArtifact); err != nil {
-		return fmt.Errorf("Failed to export environment variable: %s", apkEnvKey)
+
+	// Use the correct env key for the selected build type
+	envKey := map[bool]string{true: apkEnvKey, false: aabEnvKey}[config.BuildType == "apk"]
+	if err := tools.ExportEnvironmentWithEnvman(envKey, lastExportedArtifact); err != nil {
+		return fmt.Errorf("Failed to export environment variable: %s", envKey)
 	}
-	log.Printf("  Env    [ $%s = $BITRISE_DEPLOY_DIR/%s ]", apkEnvKey, filepath.Base(lastExportedArtifact))
+	log.Printf("  Env    [ $%s = $BITRISE_DEPLOY_DIR/%s ]", envKey, filepath.Base(lastExportedArtifact))
 
 	var paths, sep string
 	for _, path := range exportedArtifactPaths {
 		paths += sep + "$BITRISE_DEPLOY_DIR/" + filepath.Base(path)
 		sep = "| \\\n" + strings.Repeat(" ", 11)
 	}
 
-	if err := tools.ExportEnvironmentWithEnvman(apkListEnvKey, strings.Join(exportedArtifactPaths, "|")); err != nil {
-		return fmt.Errorf("Failed to export environment variable: %s", apkListEnvKey)
+	// Use the correct env key for the selected build type
+	envKey = map[bool]string{true: apkListEnvKey, false: aabListEnvKey}[config.BuildType == "apk"]
```

Comment:
Do not feel this is worth (in line savings) it in contrast to an if/switch

Reply:
Fixed

Reply:
👍